service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ChildControllerMidEventTest.java

```diff
@@ -0,0 +1,219 @@
+    }
+
+    @Test
+    void shouldReturnNoDateOfBirthErrorsForNewChildrenWhenValidDateOfBirth() throws Exception {
+        CallbackRequest request = CallbackRequest.builder()
+            .caseDetails(CaseDetails.builder()
+                .id(12345L)
+                .data(ImmutableMap.of(
+                    "children1", ImmutableList.of(
+                        ImmutableMap.of(
+                            "id", "",
+                            "value", Child.builder()
+                                .party(ChildParty.builder()
+                                    .dateOfBirth(Date.from(ZonedDateTime.now().minusDays(1).toInstant()))
+                                    .build())
+                                .build()
+                        )
+                    )
+                ))
+                .build())
+            .build();
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse = makeRequest(request);
+
+        assertThat(callbackResponse.getErrors()).isEmpty();
+    }
+
+    @Test
+    void shouldReturnNoDateOfBirthErrorsWhenCaseDataIsEmpty() throws Exception {
+        CallbackRequest request = CallbackRequest.builder()
+            .caseDetails(CaseDetails.builder()
+                .id(12345L)
+                .data(ImmutableMap.of())
+                .build())
+            .build();
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse = makeRequest(request);
+
+        assertThat(callbackResponse.getErrors()).isEmpty();
+    }
+
+    private OldChild createChild(ZonedDateTime dateOfBirth) {
+        return OldChild.builder()
+            .childDOB(Date.from(dateOfBirth.toInstant()))
+            .build();
+    }
+
+    private AboutToStartOrSubmitCallbackResponse makeRequest(OldChildren children) throws Exception {
+        HashMap<String, Object> map = MAPPER.readValue(MAPPER.writeValueAsString(children),
+            new TypeReference<Map<String, Object>>() {
+            });
+
+        CallbackRequest request = CallbackRequest.builder()
+            .caseDetails(CaseDetails.builder()
+                .id(12345L)
+                .data(ImmutableMap.<String, Object>builder().put("children", map).build())
+                .build())
+            .build();
+
+        MvcResult response = mockMvc
+            .perform(post("/callback/enter-children/mid-event")
+                .header("authorization", AUTH_TOKEN)
+                .header("user-id", USER_ID)
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(MAPPER.writeValueAsBytes(request)))
+            .andExpect(status().isOk())
+            .andReturn();
+
+        return MAPPER.readValue(response.getResponse()
+            .getContentAsByteArray(), AboutToStartOrSubmitCallbackResponse.class);
+    }
+
+    private AboutToStartOrSubmitCallbackResponse makeRequest(CallbackRequest request) throws Exception {
+        MvcResult response = mockMvc
```

Comment:
Repeated core of making response, could be extracted, probably it's repeated across the board isn't it?

Reply:
I am not inclined to fix within this PR

Reply:
That's fine, something to keep it in mind when we work in the area!